spyder/plugins/ipythonconsole/widgets/main_widget.py

```diff
@@ -1851,9 +1851,10 @@ def create_client_for_file(self, filename, is_cython=False):
         self.master_clients -= 1
 
         # Rename client tab with filename
-        client.allow_rename = False
-        tab_text = self.disambiguate_fname(filename)
-        self.rename_client_tab(client, tab_text)
+        if (client):
```

Comment:
```suggestion
        if client is not None:
```

According to issue #21884, the problem happens when we reach this point and `client` is `None`. So we can improve a little bit this validation.